### PR TITLE
[WIP] test: add getaddress mock

### LINKED ISSUE #398: Remove network dependency in get_address_test
## Describe the bug

`get_address` internally uses of `getaddrinfo` to get hints to use either IPv4 or IPv6. While this is fine to use in production, its not a good practice to use in tests since it might fail due to "outside" conditions. For example, if there is no dns resolution, the `get_address_test` fails due to a timeout as can be seen in the following output: 

```bash
get_address_test time out (After 30 seconds)
11/11 get_address_test     TIMEOUT        30.04s   killed by signal 15 SIGTERM
>>> MALLOC_PERTURB_=154 /builddir/build/BUILD/hirte-0.4.0/redhat-linux-build/src/libhirte/test/common/network/get_address_test
Summary of Failures:
11/11 get_address_test     TIMEOUT        30.04s   killed by signal 15 SIGTERM
```

## To Reproduce

Running `meson test -C builddir` with no dns resolution active on the machine. 

## Expected behavior

Either a passing or disabled `get_address_test`. 

## Proposal to fix

Create a type that is matching the `getaddrinfo` and having it being passed in as function pointer. This way the caller can determine the behavior of the function. For the production code, just pass the original `getaddrinfo`. For the test code, just implement a dummy function that sets the `res->ai_family` and `res->ai_addr` and pass it to the call. 
If that is not possible or too complicated, lets deactivate the test for now (removing from the list in meson.build).

### PR DESCRIPTION
**Pull request by dougsland**
There are test pipelines environments that cannot use external network due security restrictions. This patch ensure the code can still be tested.

Fixes: https://github.com/containers/hirte/issues/398

Signed-off-by: Douglas Schilling Landgraf <dougsland@redhat.com>
Signed-off-by: Michael Engel <mengel@redhat.com>

### CODE REVIEW

**Comment by dougsland** [comment]
Adding @engelmi as co-author as he spend time with me explaining his ideas/code. :)

File: src/libhirte/test/common/network/get_address_test.c [hunk 1]
```diff
@@ -22,13 +57,15 @@ bool test_get_address_failure(const char *domain) {
                 fprintf(stdout, "FAILED: get_address('%s') - Expected null, but got non-null (%s)\n", domain, ip);
                 return false;
         }
+
+
```
Comment (mwperina) [inline src/libhirte/test/common/network/get_address_test.c]: What are those 2 empty lines for?

File: src/libhirte/test/common/network/get_address_test.c [hunk 2]
```diff
@@ -7,10 +7,45 @@
 #include "libhirte/common/network.h"
 #include "libhirte/common/string-util.h"
 
+int getaddrinfo_func_mock_failure(const char *, const char *, const struct addrinfo *, struct addrinfo **) {
+        return -1;
+}
+
+int getaddrinfo_func_mock(const char *, const char *, const struct addrinfo *, struct addrinfo **foo) {
```
Comment (mwperina) [inline src/libhirte/test/common/network/get_address_test.c]: Just wondering if it wouldn't be better to have more complete mock definition?

1. Have a mocking parameters and results structure:
````
struct getaddrinfo_mock_params {
    char* hostname;
    char** resolved_addresses;
    int result;
}
````
2. Initialize this structure for all test cases:
````
static struct getaddrinfo_mock_params[] test_data = [
    { NULL, NULL, -1 },
    { ".", NULL, -1 },
    { "redhat", NULL, -1 },
    { "?10.10.10.3", NULL -1 },
    { "192.168.1.", NULL, -1 },
    { "8.8.8.8", ["8.8.8.8"], 0},
    { "localhost.localdomain", ["127.0.01", "::1"], 0}
];
````
3. Having only `getaddrinfo_func_mock`, which uses `test_data` array and create output according to the input and `test_data` content

File: src/libhirte/test/common/network/get_address_test.c (hunk 1 above)
Comment (dougsland) [reply src/libhirte/test/common/network/get_address_test.c]: nothing, it was late in the night ;-)

**Comment by dougsland** [comment]
ok, I mess up the rebase locally and the push is showing more patches than needed. Let me start a fresh PR.

**Pull request closed without merging**